semantic-analysis/src/Analysis/Syntax.hs

```diff
@@ -102,26 +114,28 @@ parseFile path = do
     -- FIXME: this should use the span of the source file, not an empty span.
     Right (_, Just root) -> pure (File (Ref.fromPath (Path.absRel path)) root)
 
-parseGraph :: A.Value -> A.Parser (IntMap.IntMap Term, Maybe Term)
+newtype Graph = Graph { terms :: IntMap.IntMap Term }
```

Comment:
For clarity. This is the adjacency list for the syntax tree; technically it's always tree-shaped, but we have no guarantee of ordering on when the nodes are deserialized, and even if we had one, the general case—for arbitrary `tree-sitter-graph` programs—is not guaranteed to be acyclic. (Thus our use of `Graph -> Term` during the decoding process, below.)